src/mostlyai/qa/report.py

```diff
@@ -225,6 +223,11 @@ def report(
         )
         on_progress(current=30, total=100)
 
+        # ensure that embeddings are all of equal size for a fair 3-way comparison
+        max_sample_size_embeddings = min(10_000_000, syn_sample_size, trn_sample_size)
```

Comment:
Why bother about upper-bound of `10_000_000` at all? Looks a bit random, even more so, because it's not a constant.